src/mon/MgrMonitor.cc

```diff
@@ -1012,13 +1012,24 @@ bool MgrMonitor::preprocess_command(MonOpRequestRef op)
       f->dump_object("mgrmap", m);
     }
     f->flush(rdata);
-  } else if (prefix == "mgr module ls") {
+} else if (prefix == "mgr module ls") {
     if (f) {
+      // Create a mapping from module names to ModuleInfo pointers
+      std::unordered_map<std::string, const MgrMap::ModuleInfo*> module_info_map;
```

Comment:
You are manipulating pointers into a vector, if I'm not mistaken. I do not know the locking infra for the modules. Are we sure there is no possibility of change to the modules vector by another thread during the execution of this function?

Reply:
Yeah. How about dropping the intermediary container and use e.g. `std::find_if` instead?